dataprofiler/profilers/numerical_column_stats.py

```diff
@@ -950,6 +955,162 @@ def _get_percentile(self, percentiles):
             quantiles[499] = median_value
         return quantiles
 
+    @BaseColumnProfiler._timeit(name="median_absolute_deviation")
+    def _get_median_abs_dev(self):
+        """
+        Get median absolute deviation.
+
+        :return: median absolute deviation
+        """
+        if not self._has_histogram:
+            return np.nan
+
+        bin_counts = self._stored_histogram['histogram']['bin_counts']
+        bin_edges = self._stored_histogram['histogram']['bin_edges']
+
+        # normalize bin counts
+        bin_counts = bin_counts.astype(float)
+        normalized_bin_counts = bin_counts / np.sum(bin_counts)
+
+        # calculate deviations from median
+        median = self.quantiles[499]
+        bin_edges = bin_edges - median
+
+        # find the break point to fold the deviation list
+        id_zero = None
+        for i in range(len(bin_edges)):
+            if bin_edges[i] > 0:
+                id_zero = i
+                break
+
+        # if all bin edges are positive (no break point),
+        # interpolate the histogram to find the median value
+        if id_zero is None:
+            bin_counts = bin_counts.astype(float)
+            normalized_bin_counts = bin_counts / np.sum(bin_counts)
+            cumsum_bin_counts = np.cumsum(normalized_bin_counts)
+
+            median_bin_inds = cumsum_bin_counts == 0.5
```

Comment:
why do we need to check this if we already interp below?

Reply:
this is to interpolate in case we have only positive or negative histogram. So, we don't have to go through the long loop below it. Btw, the median should be 0 in this case, which has been updated.